ext/hal/st/lib/sensor/vl53l1x/README

```diff
@@ -0,0 +1,53 @@
+VL53L1X
+#######
+
+Origin:
+   ST Microelectronics
+   http://www.st.com/en/embedded-software/stsw-img007.html
+
+Status:
+   version 2.3.3
+
+Purpose:
+   ST Microelectonics official API to use vl53l1x sensor.
+
+Description:
+   This library written by STMicroelectronics is dedicated to vl53l1x time of flight sensor.
+   Instead of providing a full description of the registers, ST decided to provide a C library to drive this sensor.
+
+   This library is called from the vl53l1x driver and is kept transparent for user.
+   User is calling a standard Zephyr driver and then, the driver makes some calls to this library.
+
+   In order to fit with Zephyr rules and simplify version updates, we have done a minimum of modifications :
+      - keep only API directory
+        (remove documentation and samples, see ST website for this)
+      - rename Api to api
+      - customize api/platform to fit in Zephyr (mainly for I2C access)
+        vl53l1x_platform.c and other platform related files have been moved to Zephyr drivers directory.
+
+   In order to use this library, you have to :
+      * define CONFIG_HAS_STLIB and CONFIG_VL53L1X
+      * use NEWLIB_LIBC in prj.conf (needed for abs function)
```

Comment:
I'm currently adding a similar external hal from ST Microelectronics for other sensors. I also need to
define NEWLIB_LIBC, but instead of letting the user to define it in prj.conf I used imply in Kconfig of the external
code. Have you considered to do the same?
See #14111 and in particular a40e62975 commit.

Reply:
yes, sound better. 
wait a moment, I will push again.

Reply:
OK, just remember to squash together similar commits as I was saying before.
I suggest to save the branch before squashing just in case